inst/tests/tests.Rraw

```diff
@@ -16740,3 +16740,10 @@ test(2131, lapply(x[ , list(dt = list(.SD)), by = a]$dt, attr, '.data.table.lock
 ########################
 #  Add new tests here  #
 ########################
+
+## names(.SD) - issue #795
+DT <- data.table(a=1:6, b=1:6, c=rep(c(T,F), 3))
+mycols <- 1:2
+test(2131.1, DT[, .SD := lapply(.SD, `*`, 2), .SDcols = mycols], data.table(a = (1:6)*2, b = (1:6)*2, c = rep(c(T, F), 3)))
+test(2131.3, DT[, .SD := lapply(.SD, '*', 2), .SDcols = -3L], data.table(a = (1:6)*4, b = (1:6)*4, c = rep(c(T, F), 3)))
+test(2131.5, DT[, .SD := lapply(.SD, '*', 2)], data.table(a = (1:6)*8, b = (1:6)*8, c = rep(c(T, F), 3) * 2))
```

Comment:
Multiply logical vector is quite an unfortunate use case because, unless multiplied by 0, it won't change the value. Much better to test `mycost = 2:3` and then we can ensure that first column is not being changed.

Reply:
Multiplying a logical vector by a number changes the value on my computer: 
```
data.table(a = c(TRUE, FALSE))[, a := 2 * a][]
       a
   <num>
1:     2
2:     0
```
I agree that the tests needed improved and made some changes.